services/secondary-storage/server/test/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResourceTest.java

```diff
@@ -18,91 +18,68 @@
-
-        // attempt a configured mount
-        final Map<String, Object> params = PropertiesUtil.toMap(loadProperties());
-        String sampleMount = (String)params.get("testCifsMount");
-        if (!sampleMount.isEmpty()) {
-            s_logger.info("functional test, mount " + sampleMount);
-            URI realMntUri = new URI(sampleMount);
-            String mntSubDir = resource.mountUri(realMntUri);
-            s_logger.info("functional test, umount " + mntSubDir);
-            resource.umount(resource.getMountingRoot() + mntSubDir, realMntUri);
-        } else {
-            s_logger.info("no entry for testCifsMount in " + "./conf/agent.properties - skip functional test");
-        }
-    }
+    @PrepareForTest(NfsSecondaryStorageResource.class)
+    public void testSwiftWriteMetadataFile() throws Exception {
+        String expected = "uniquename=test\nfilename=testfile\nsize=100\nvirtualsize=1000";
 
-    public static Properties loadProperties() throws ConfigurationException {
-        Properties properties = new Properties();
-        final File file = PropertiesUtil.findConfigFile("agent.properties");
-        if (file == null) {
-            throw new ConfigurationException("Unable to find agent.properties.");
-        }
-        s_logger.info("agent.properties found at " + file.getAbsolutePath());
-        try(FileInputStream fs = new FileInputStream(file);) {
-            properties.load(fs);
-        } catch (final FileNotFoundException ex) {
-            throw new CloudRuntimeException("Cannot find the file: " + file.getAbsolutePath(), ex);
-        } catch (final IOException ex) {
-            throw new CloudRuntimeException("IOException in reading " + file.getAbsolutePath(), ex);
-        }
-        return properties;
+        StringWriter stringWriter = new StringWriter();
+        BufferedWriter bufferWriter = new BufferedWriter(stringWriter);
+        PowerMockito.whenNew(BufferedWriter.class).withArguments(any(FileWriter.class)).thenReturn(bufferWriter);
+
+        resource.swiftWriteMetadataFile("testfile", "test", "testfile", 100, 1000);
+
+        Assert.assertEquals(expected, stringWriter.toString());
     }
 
+    @Test
+    public void testCleanupStagingNfs() throws Exception{
+
+        NfsSecondaryStorageResource spyResource = spy(resource);
+        RuntimeException exception = new RuntimeException();
+        doThrow(exception).when(spyResource).execute(any(DeleteCommand.class));
+        TemplateObjectTO mockTemplate = Mockito.mock(TemplateObjectTO.class);
+
+        TestAppender.TestAppenderBuilder appenderBuilder = new TestAppender.TestAppenderBuilder();
+        appenderBuilder.addExpectedPattern(Level.DEBUG, "Failed to clean up staging area:");
+        TestAppender testLogAppender = appenderBuilder.build();
+        Logger testLogger = Logger.getLogger(NfsSecondaryStorageResource.class);
+        TestAppender.safeAddAppender(testLogger, testLogAppender);
```

Comment:
Minor nit (implement if you like): Lines 74-78 could be reduced to the following:
``` 
testAppender = new TestAppender.TestAppenderBuilder()
                             .addExpectedPattern(Level.DEBUG, "Failed to clean up staging area:")
                             .build();
TestAppender.safeAddAppender(Logger.getLogger(NfsSecondaryStorageResource.class), testAppender);
```